lib/logstash/outputs/s3.rb

```diff
@@ -252,13 +269,36 @@ def restore_from_crashes
     end
   end
 
+  public 
+  def shouldcleanup(prefix)
+     return @empty_uploads[prefix] > @no_event_wait
+  end
```

Comment:
Since this is a predicate method It should contains an integration mark.
I also suggest we rename it to `need_cleanup?(prefix)`  since it express more necessity.